### DISCUSSION TITLE
JSON response wrapped in square brackets.

### DISCUSSION THREAD

**Discussion by K0styan** [supabase/supabase#8736] [Questions] [answered]
I have a Supabase hosted database.
When I send API requests(in Postman or in console) to this database, I got a data but it is wrapped in square brackets.
For example:
```json
[
    {
        "id": 1,
        "created_at": "2022-08-17T16:10:18+03:00",
        "clubOwner": "e192e097-7bb0-45ff-900f-0653725469bb",
        "clubName": "Club 1",
        "clubAddress": "club address",
        "clubCourtsHard": 6,
        "clubCourtsClay": 6,
        "clubCourtsAiGrass": 0,
        "clubCourtsGrass": 0
    }
]
```
Why the brackets are added to a response? How receive responses without these square brackets?

**Comment by dshukertjr** [comment, marked as answer]
By default, Postgrest, rest API engine that Supabase uses behind the scene, returns data in array format since there might be more than 1 data being returned. 

If you know that you will only receive one row, you can use the [`.single()`](https://github.com/supabase/postgrest-js/blob/master/src/lib/PostgrestTransformBuilder.ts#L100) method using the client, or add `Accept=application/vnd.pgrst.object+json` header to your request. This will remove the square brackets. Note that with this, the request will fail if you have 0 or more than one returned data.
